File: 0581-shortest-unsorted-continuous-subarray/0581-shortest-unsorted-continuous-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findUnsortedSubarray = function(nums) {
    const sorted = nums.slice()
    sorted.sort((a, b) => a - b)
    let start = nums.length-1, end = 0 
    for (let i=0; i < nums.length; i++) {
        if (sorted[i] !== nums[i]) {
            start = Math.min(start, i)
            end = Math.max(end, i)
        }
    }
    return end - start > 0 ? end - start + 1 : 0
};

File: 0581-shortest-unsorted-continuous-subarray/0581-shortest-unsorted-continuous-subarray.py
class Solution:
    def findUnsortedSubarray(self, nums: List[int]) -> int:
        sorted_nums, lo, hi = sorted(nums), len(nums), 0
        for i in range(0, len(nums)):
            if sorted_nums[i] != nums[i]:
                lo = min(lo, i)
                hi = max(hi, i)
        return hi - lo + 1 if hi - lo > 0 else 0
        